bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt

```diff
@@ -1,55 +1,136 @@
 Apache BookKeeper
-Copyright 2011-2017 The Apache Software Foundation
+Copyright 2011-2018 The Apache Software Foundation
 
-Licensed under the Apache License, Version 2.0 (the
-"License"); you may not use this file except in compliance
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+------------------------------------------------------------------------------------
+- lib/io.dropwizard.metrics-metrics-core-3.1.0.jar
```

Comment:
It is okay and enough to keep versioning in LICENSE. However I would suggest removing versioning from NOTICE file: 1) versioning is not legally required in NOTICE. 2)  NOTICE as short as possible as it will affect downstream projects. this is going to cause NOTICE updates everytime we bumped version.

 http://www.apache.org/dev/licensing-howto.html#mod-notice

Reply:
the versions are there to facilitate automated checking. if we remove the versions we can't ensure that what is in the notice file exists in the shipped tarball. It will need to be checked manually, so more chance of a -1 on a release candidate.

Also, notices can change between version, so the rationale for having them in the LICENSE holds for the NOTICE also.

Reply:
People will do the bare minimum to make their builds pass. If there are no versions in the NOTICE, they won't touch the notice, and we'll end up with -1 on release candidates.

This notice should have no effect on downstream projects. Downstream depend on our maven jar, not our binary tarballs, which is covered by the top level NOTICE.

Reply:
@ivankelly fine with me.